### DISCUSSION TITLE
Use array key as value

### DISCUSSION THREAD

**Discussion by CaaioSB** [react-hook-form/react-hook-form#3291] [Q&A] [answered]
### Imagine the JSON below:
```json
const fieldProps = {
   "if":{
      "client_relationship":[
         "taker",
         "guarantor",
         "other"
      ]
   }
}
```
### What I want to do:
Basically define the key `client_relationship` by an Input / Select.

My result so far:
![image](https://user-images.githubusercontent.com/34246280/97467794-507f4580-1923-11eb-8757-663f5bd379d0.png)

### Code snippet
![image](https://user-images.githubusercontent.com/34246280/97468419-05196700-1924-11eb-8604-4e5d37992300.png)

### Conclusion
So in this example, I would change the select item to `client_relationship2` and json would become this:

```json
{
   "if":{
      "client_relationship2":[
         "taker",
         "guarantor",
         "other"
      ]
   }
}
```

**Comment by bluebill1049** [comment, marked as answer]
what's the issue here?

**Reply by CaaioSB** [reply]
i just want know if i can change a json key by input

**Reply by bluebill1049** [reply]
```
<input name="test.bill" />
<input name="test.bill1" />
```

?

**Reply by CaaioSB** [reply]
on my scenary don't work, but thank you :) 
my form is totally dynamic, i use react hook form for change components properties on my form builder, so the user i should be able to change the key, but thank you very much

**Reply by CaaioSB** [reply]
you can delete this discussion? thx and sry
